internal/pkg/cli/app_delete.go

```diff
@@ -0,0 +1,309 @@
+	}
+
+	if !deleteConfirmed {
+		return errAppDeleteCancelled
+	}
+
+	return nil
+}
+
+func (opts *deleteAppOpts) init() error {
+	projectService, err := store.New()
+	if err != nil {
+		return fmt.Errorf("create project service: %w", err)
+	}
+	opts.projectService = projectService
+
+	workspaceService, err := workspace.New()
+	if err != nil {
+		return fmt.Errorf("intialize workspace service: %w", err)
+	}
+	opts.workspaceService = workspaceService
+
+	return nil
+}
+
+func (opts *deleteAppOpts) sourceInputs() error {
+	if opts.ProjectName() == "" {
+		return errNoProjectInWorkspace
+	}
+
+	if err := opts.sourceProjectData(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (opts *deleteAppOpts) sourceProjectData() error {
+	if err := opts.sourceProjectApplications(); err != nil {
+		return err
+	}
+
+	if err := opts.sourceProjectEnvironments(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (opts *deleteAppOpts) sourceProjectApplications() error {
+	apps, err := opts.workspaceService.Apps()
+
+	if err != nil {
+		return fmt.Errorf("get app names: %w", err)
+	}
+
+	if len(apps) == 0 {
+		// TODO: recommend follow up command - app init?
+		return errors.New("no applications found")
+	}
+
+	var appNames []string
+	for _, app := range apps {
+		appNames = append(appNames, app.AppName())
+	}
+
+	opts.localProjectAppNames = appNames
+
+	return nil
+}
+
+func (opts *deleteAppOpts) sourceProjectEnvironments() error {
+	envs, err := opts.projectService.ListEnvironments(opts.ProjectName())
+
+	if err != nil {
+		return fmt.Errorf("get environments: %w", err)
+	}
+
+	if len(envs) == 0 {
+		// TODO: recommend follow up command - env init?
+		log.Infof("couldn't find any environments associated with project %s, try initializing one: %s\n",
+			color.HighlightUserInput(opts.ProjectName()),
+			color.HighlightCode("archer env init"))
+
+		return errors.New("no environments found")
+	}
+
+	opts.projectEnvironments = envs
+
+	return nil
+}
+
+func (opts deleteAppOpts) deleteApp() error {
+	if err := opts.deleteStacks(); err != nil {
+		return err
+	}
+	if err := opts.emptyECRRepos(); err != nil {
+		return err
+	}
+	if err := opts.removeAppProjectResources(); err != nil {
+		return err
+	}
+	if err := opts.deleteSSMParam(); err != nil {
+		return err
+	}
+	if err := opts.deleteWorkspaceFile(); err != nil {
+		return err
+	}
+
+	log.Ssuccessf("removed app %s from project %s\n", opts.app, opts.projectName)
+
+	return nil
+}
+
+func (opts deleteAppOpts) deleteStacks() error {
+	for _, env := range opts.projectEnvironments {
+		sess, err := session.FromRole(env.ManagerRoleARN, env.Region)
+		if err != nil {
+			return err
+		}
+
+		cfClient := cloudformation.New(sess)
+
+		stackName := fmt.Sprintf("%s-%s-%s", opts.projectName, env.Name, opts.app)
+
+		opts.spinner.Start(fmt.Sprintf("deleting app %s from env %s", opts.app, env.Name))
+		if err := cfClient.DeleteStackAndWait(stackName); err != nil {
```

Comment:
I personally don't think so. I could see value in a layer between workflow code and CloudFormation code that orchestrates dependencies such as CloudFormation, but putting an `archer` package dependency on `cloudformation` doesn't feel right.

Reply:
From my point of view, we can have a `aws/cloudformation` package that wraps the SDK's package for testing (with shallow wrappers like `DeleteStack`).
However, the `deploy/cloudformation` package is meant to be a deeper abstraction that work on top of our concepts. At least that's how I saw it :P

Reply:
I think we're both advocating for the same thing. The `deploy/cloudformation` package is currently doing a bit of both shallow wrapping (considering it's using the `service/cloudformation` client) and archer abstractions. I can move these out to `aws/cloudformation`, but it will result in some temporary duplication.

I didn't feel much of a need for a deeper abstraction in this specific case.